Added Download Button

dimi-23 opened the pull request:
**Fixes issue:**
<!-- [Mention the issue number it fixes or add the details of the changes if it doesn't has a specific issue. -->
#59 

**Changes:**
<!-- Add here what changes were made in this awesome pull request. -->

A download button has been added to download the code easily.

![image](https://user-images.githubusercontent.com/28829807/36614830-fcdeb432-1903-11e8-92f0-73e2f6c19830.png)


<!-- Make sure that you enjoyed being the part of the OpenGenus Community. We would love to hear how can we make your contributing experience better. Thank you. Have a nice day! -->

Sakshi183 commented:
This is an offline chrome extension .So, all the algorithms are already available offline.So, I guess the option to download code is not required. Instead, we can open the directory where the code is stored for editing the code.

dimi-23 commented:
Hi @Sakshi183 

Yes, Quark is offline but I added the feature keeping in mind the following points:

1. Codes available are accessible only from the Chrome browser. If the users are given an option to save it to their desired directory, it will help them when the browser is not accessible. 

2. Accessing extension's directory is a tedious job and no user will do that. The users can copy-paste the code if they need it to run on their local compilers, but if a download option is present it can make the process faster.

3. If the download option is not directly available and the users try to access the codes from their default directory, they may mishandle and misplace some codes, resulting in undesirable behavior of the extension.

So, keeping in mind these points this issue was raised and implemented.

@AdiChat please correct me if I am wrong in my assessment.

AdiChat commented:
Yes, a download feature is useful. Users do have a tendency to download code and work on it locally. :+1:

AdiChat commented on code/js/popup.js:
```diff
@@ -154,9 +154,9 @@ function dumpBookmarks(query)
 	
 						   	if(!favs.includes(fname)) {
-						   		inside_text = inside_text + "<a  target='_blank' href='/code/"+key+"/"+obj[key][dd]+"'>"+sub_result_number+". "+obj[key][dd]+"</a>"+"&nbsp;&nbsp;<i id='myStar"+temp2+"\' class='fa fa-star'></i><br>";
+						   		inside_text = inside_text + "<a  target='_blank' href='/code/"+key+"/"+obj[key][dd]+"'>"+sub_result_number+". "+obj[key][dd]+"</a>"+"&nbsp;&nbsp;<a style='color:inherit' href='/code/"+key+"/"+obj[key][dd]+"'download><i id='myDownload"+temp2+"\' style='float:right' class='fa fa-download'></i></a><i id='myStar"+temp+"\' style='float:right;width:8%' class='fa fa-star'></i><br>";
```
It looks like `temp` should be replaced by `temp2`. This is breaking the favorites features. Same is applicable to the else section. :+1: 

Kindly take a look into this. :+1:

dimi-23 replied:
Oops yes! Made the corresponding changes.

AdiChat commented:
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**
